src/AuthState.ts

```diff
@@ -29,9 +38,33 @@ export interface AuthState {
     /**
      * Was there a signin or silent renew error?
      */
-    error?: Error;
+    error?: ErrorContext;
 }
 
+/**
+ * Represents an error while execution of a signing, renew, ...
+ *
+ * @public
+ */
+export type ErrorContext = {
```

Comment:
Why not re-using the Error interface? Something like:
`type ErrrorWithSource = Error & ({ source: "signinCallback" }...`

Passing the `error` into that type can be done with the spread operator:
```ts
 error: {
  ...error,
  source: xxxx
}
```

Another benefit, it would be an extension and not something that breaks existing apps...

Reply:
This was my first intention, too. But it breaks on Internet Explorer as it not an interface there, unfortunately. As still many people are using the Internet Explorer (in cooperates) I don't want to exclude them for now. :-/

Reply:
This library does not support Internet Explorer since the start. Specifically the underlying oidc-client-ts also not. I am working in a cooperate and people here are using either Edge, Chrome (most) or Firefox since years.

That said, we can improve the detection of `Error`, but lets use that interface as basis...

Reply:
Ok. I'll adjust.

Reply:
Adjusted as purposed.